packages/odyssey-react/src/components/Banner/index.tsx

```diff
@@ -0,0 +1,126 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from 'react';
+import type { FunctionComponent, ReactNode } from 'react';
+import { useCx, useOmit } from '../../utils';
+import Button from '../Button';
+
+export type BannerVariants = 'info' | 'danger' | 'caution';
+
+type ComponentProps = {
+  /**
+   * Actions, or links to be rendered on the right side of 
+   * the component.
+   */
+  children: ReactNode,
+  
+  /**
+   * The visual variant to be displayed to the user.
+   * @default info
+   */
+  variant?: BannerVariants,
+
+  /**
+   * Human-readable title for the banner.
+   */
+  title: string;
+
+  /**
+   * Human-readable descriptive content for the banner.
+   */
+  content: string;
+
+  /**
+   * Determines whether the banner should be displayed to the user.
+   */
+  visible: boolean,
+
+  /**
+   * function callback which enables the display of the dismiss button.
+   */
+  onDismiss?: () => void,
+
+  /**
+   * Applies an aria-label to the dismiss button. This is required if an 
+   * onDismiss callback is provided.
```

Comment:
would be good to enforce this through types

Reply:
Done, inspired by [this article](https://www.benmvp.com/blog/conditional-react-props-typescript/) shared by @patrickmcdougle-okta! (Thank you!)